sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -503,13 +503,32 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
     }
   }
 
+  private def convertTypedLiteralToString(literal: Literal): String = literal match {
+    case Literal(data: Int, _: DateType) =>
+      UTF8String.fromString(
+        DateFormatter(getZoneId(SQLConf.get.sessionLocalTimeZone))
+          .format(data)).toString
+    case Literal(data: Long, _: TimestampType) =>
+      UTF8String.fromString(
+        TimestampFormatter.getFractionFormatter(getZoneId(SQLConf.get.sessionLocalTimeZone))
+          .format(data)).toString
+    case Literal(data: CalendarInterval, _: CalendarIntervalType) =>
+      UTF8String.fromString(data.toString).toString
```

Comment:
```suggestion
    case Literal(data: Int, _: DateType) =>
      DateFormatter(getZoneId(SQLConf.get.sessionLocalTimeZone)).format(data)
    case Literal(data: Long, _: TimestampType) =>
      TimestampFormatter.getFractionFormatter(getZoneId(SQLConf.get.sessionLocalTimeZone))
        .format(data)
    case Literal(data: CalendarInterval, _: CalendarIntervalType) => data.toString
```

Reply:
Updated

Reply:
@MaxGekk  Use BinaryType as partition column in DataSource API have bug so I remove test case of binary type. I will raise a issue later  about this bug.

Reply:
Add. back using csv instead now.
also cc @maropu @cloud-fan